setup.py

```diff
@@ -1,94 +1,56 @@
-
-
-@contextlib.contextmanager
-def cwd(path):
-    curdir = os.getcwd()
-    try:
-        os.chdir(path)
-        yield
-    finally:
-        os.chdir(curdir)
-
-# inspired by https://blog.shazam.com/python-microlibs-5be9461ad979
-
-
-packages = ['vaex-core', 'vaex-viz', 'vaex-hdf5', 'vaex-server', 'vaex-astro', 'vaex-jupyter', 'vaex-ml', 'vaex-meta', 'vaex-graphql', 'vaex-contrib']
-on_rtd = os.environ.get('READTHEDOCS', None) == 'True'
+# only for local development / install from source, e.g.
+# pip install https://github.com/vaexio/vaex/archive/refs/heads/master.zip
 
+from pathlib import Path
 
-class DevelopCmd(develop):
-    def run(self):
-        relative = os.path.abspath(os.path.join('packages', 'vaex-core', 'vaex'))
-        for package in packages:
-            with cwd(os.path.join('packages', package)):
-                err = os.system(f'{sys.executable} -m pip install -e .')
-                if err:
-                    raise RuntimeError(f'Oops, failed to install {package}')
-            # we need to make symbolic links from vaex-core/vaex/<name> to vaex-<name>/vaex/<name>
-            # otherwise development install do not work
-            if package not in ['vaex-core']:
-                name = package.split('-')[1]
-
-                source = os.path.abspath(os.path.join('packages', package, 'vaex', name))
-                rel_source = os.path.relpath(source, relative)
-                with cwd(relative):
-                    print('symlinking', source, name, rel_source)
-                    if os.path.exists(name) and os.readlink(name) == rel_source:
-                        print('symlink ok')
-                    else:
-                        if os.path.exists(name):
-                            print('old symlink',  os.readlink(name))
-                            os.remove(name)
-                        os.symlink(rel_source, name)
-
-
-class InstallCmd(install):
-    """ Add custom steps for the install command """
-    def run(self):
-        for package in packages:
-            with cwd(os.path.join('packages', package)):
-                os.system(f'{sys.executable} -m pip install --no-deps .')
-        for package in packages:
-            with cwd(os.path.join('packages', package)):
-                os.system(f'{sys.executable} -m pip install --upgrade .')
+from setuptools import setup
 
+packages = [
+    "vaex-core",
+    "vaex-viz",
+    "vaex-hdf5",
+    "vaex-server",
+    "vaex-astro",
+    "vaex-jupyter",
+    "vaex-ml",
+    "vaex-graphql",
+    "vaex-contrib",
+    "vaex",
+]
 
 setup(
-    name='vaex-meta',
+    name="vaex-meta",
     version="0.1.0",
     description="Convenience setup.py for when installing from the git repo",
     classifiers=[
-        'Private :: Do Not Upload to pypi server',
+        "Private :: Do Not Upload to pypi server",
     ],
     packages=[],
     install_requires=[
-        'pip',
+        f"{package}[all] @ {(Path(__file__).parent / 'packages' / package).as_uri()}"
```

Comment:
are you saying this works for an editable install? (awesome if yes!)

Reply:
nope, this doesn't work for `-e` I'm afraid.

however, I think you could do `pip install -e packages/*` if you wanted.

Reply:
In solara we do:
https://github.com/widgetti/solara/blob/master/requirements-dev.txt

But if we do this, we have to maybe update https://vaex.io/docs/installing.html